client/dashboard/data/index.ts

```diff
@@ -300,19 +300,57 @@ export const fetchSiteSettings = async ( siteIdOrSlug: string ): Promise< SiteSe
 		path: `/sites/${ siteIdOrSlug }/settings`,
 		apiVersion: '1.4',
 	} );
-	return settings;
+	return fromRawSiteSettings( settings );
 };
 
 export const updateSiteSettings = async ( siteIdOrSlug: string, data: Partial< SiteSettings > ) => {
-	return await wpcom.req.post(
+	const { updated } = await wpcom.req.post(
 		{
 			path: `/sites/${ siteIdOrSlug }/settings`,
 			apiVersion: '1.4',
 		},
-		data
+		toRawSiteSettings( data )
 	);
+	return fromRawSiteSettings( updated );
 };
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function fromRawSiteSettings( settings: any ): SiteSettings {
```

Comment:
Why are we converting the data here? Why not just interpret it in the components?

Also why disable TS?

Reply:
See https://github.com/Automattic/wp-calypso/pull/103426#discussion_r2091093907.

Reply:
I think we could define a `rest-api-types.ts` file where we define the "types" that come from the REST API if we diverge more over time.